modules/ROOT/pages/dataweave-versioning-behavior.adoc

```diff
@@ -0,0 +1,17 @@
+= Versioning Behavior in DataWeave
+
+In each version of DataWeave, changes are introduced that modify the way the language behaves. When existing scripts depend on previous behavior, system properties are introduced to toggle the change so that backwards compatibilty is not broken in the current version. These system properties that control behavior in DataWeave are called *Compatibility Flags*. Each flag defined a deletion version, after which it will no longer exists, and the behavior that it controlled will default to the new one.
+
+Specifying DataWeave's *language level* indicates to the runtime which Compatibility Flags are available. As previously stated, when upgrading to a new version there are flags that will be deleted. In order to facilitate updatating DataWeave, setting the language level of a previous version makes it so that the flags available on that version are visible, mantaining backwards compatibility. Language level defaults to the running DataWeave version and has to be at most the same as the runtime version.
+
+For mule applications the language level is determined by the app's *minMuleVersion*. Given the minMuleVersion, the language level is computed by substracting 2 major versions as shown in the following table:
+
+[%header%autowidth.spread,cols="a,a"]
+|===
+| minMuleVersion | LanguageLevel
+| 4.5 | 2.5
+| 4.4 | 2.4
+| 4.3 | 2.3
+|===
+
+For example, the `com.mulesoft.dw.xml_reader.honourMixedContentStructure` exists for language level 2.4 (it was introduced on that version and is deleted in 2.5), meaning that for DataWeave 2.5 the property will be visible only if the language level is set to 2.4, otherwise it will default to its new behavior.
```

Comment:
```suggestion
For example, the `com.mulesoft.dw.xml_reader.honourMixedContentStructure` system property exists for language level 2.4, when it was introduced, and it is removed in DataWeave 2.5. In DataWeave 2.5, the property will be visible only if you set the language level to 2.4. Otherwise, the property will default to its new behavior in 2.5.
```

Reply:
You would set minMuleVersion to 2.4 or set the %dw directive to that?

Reply:
Yeah, we should call that a compatibility flag.

The minMuleVersion is always set on the mule app and is a property of the whole application it's not just related to DW. Regarding the directive they are unrelated features, so for example if you want to migrate a mule app from 4.4.0 to 4.5.0 you can set the app language level to 2.4 so your DW app doesn't break, but you can use the `%dw 2.5` in your scripts, as 4.5.0 comes with DW 2.5.

Let me know is this clears things up

Reply:
That clarifies it for me. Thanks!